backend/app/api/ai_agent.py

```
@@ -0,0 +1,288 @@
+        )
+    if range_key == "30d":
+        return (now - timedelta(days=29)).replace(
+            hour=0, minute=0, second=0, microsecond=0
+        )
+    raise HTTPException(
+        status_code=status.HTTP_400_BAD_REQUEST,
+        detail={"code": "INVALID_RANGE", "message": "range must be one of today|7d|30d"},
+    )
+
+
+def _merge_counter(dst: dict[str, int], src: Any) -> None:
```

Ruff (0.15.10)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `src`

(ANN401)